### PR TITLE
simplify the code for diagnostics severity adjustment

### PR DESCRIPTION
**Pull request by Martin521**
## Description

This PR simplifies the code for adjusting diagnostics severity, without change to the functionality.
It will be needed for "scoped nowarn", where the adjustment logic becomes even more complex, but makes sense also in itself.

### CODE REVIEW

**Comment by Martin521** [comment]
Can somebody add the "no release notes" label?
Thanks

**Comment by psfinaki** [comment]
There you go, Martin :) Thanks for splitting the refactoring into a separate PR, that's always helpful.

File: src/Compiler/Symbols/FSharpDiagnostic.fs [hunk 1]
```diff
@@ -319,23 +318,18 @@ type internal CompilationDiagnosticLogger (debugName: string, options: FSharpDia
...
-             diagnostic.ReportAsInfo (options, severity) then 
+        match diagnostic.AdjustedSeverity(options, severity) with
+        | FSharpDiagnosticSeverity.Hidden -> []
+        | sev ->
 
             // We use the first line of the file as a fallbackRange for reporting unexpected errors.
             // Not ideal, but it's hard to see what else to do.
             let fallbackRange = rangeN mainInputFileName 1
-            let diagnostic = FSharpDiagnostic.CreateFromExceptionAndAdjustEof (diagnostic, severity, fallbackRange, fileInfo, suggestNames, flatErrors, symbolEnv)
+            let diagnostic = FSharpDiagnostic.CreateFromExceptionAndAdjustEof (diagnostic, sev, fallbackRange, fileInfo, suggestNames, flatErrors, symbolEnv)
```
Comment (KevinRansom) [inline src/Compiler/Symbols/FSharpDiagnostic.fs:328]: It was originally severity, we have always had problems with abbreviations in this codebase, in general avoid them, especially if they were originally unabbreviated.

File: src/Compiler/Driver/CompilerDiagnostics.fs [hunk 2]
```diff
@@ -409,47 +409,25 @@ type PhasedDiagnostic with
...
-
-        | FSharpDiagnosticSeverity.Warning -> x.IsEnabled(severity, options) && not (List.contains x.Number options.WarnOff)
-
-        // Informational become warning if explicitly on and not explicitly off
-        | FSharpDiagnosticSeverity.Info ->
-            let n = x.Number
-            List.contains n options.WarnOn && not (List.contains n options.WarnOff)
-
-        | FSharpDiagnosticSeverity.Hidden -> false
+    member x.AdjustedSeverity(options, severity) =
```
Comment (KevinRansom) [inline src/Compiler/Driver/CompilerDiagnostics.fs:412]: Should it be AdjustSeverity?

**Review by KevinRansom — COMMENTED** [review]
Thank you, nice bit of cleanup.
A couple of nits, but other than that this is ready to go.

File: src/Compiler/Symbols/FSharpDiagnostic.fs (hunk 1 above)
Comment (Martin521) [reply src/Compiler/Symbols/FSharpDiagnostic.fs:328]: I didn't want to shadow the original severity. I'll make it `adjustedSeverity`.

**Review by vzarytovskii — APPROVED** [review]

**Review by edgarfgp — APPROVED** [review]
Thanks Martin 👍🏻

**Pull request merged**